content/blog/2019/07/2019-07-15-pipeline-config-history-plugin.adoc

```diff
@@ -0,0 +1,88 @@
+  These files are also saved in the SCM.
+  Even so a Jenkinsfile is still needed (or a pipeline script in the job).
+
+With every save operation of the job configuration, JobConfigHistory creates a copy of the actual job configuration if something has changed.
+For pipeline jobs that only works if the pipeline configuration is inserted in the job configuration page as script.
+Jenkinsfiles or shared libraries cannot get detected by JobConfigHistory.
+You have to go to the SCM to view changes of the Jenkinsfiles or shared libraries.
+It is complex and time intensive to find a correlation between the time of a build and a change to the Jenkinsfile or shared library.
+
+This new problem cannot get solved by JobConfigHistory and a new solution were needed to get found, that also detects these changes and provides a way to view these changes within the Jenkins GUI.
+So we developed Pipeline Configuration History.
+
+During every pipeline run the Jenkinsfile and used shared libraries get saved in the `builds` directory of the job.
+The function of Pipeline Configuration History is to check after every pipeline run, if there were changes of the pipeline files between the last and the previous run and if so, it saves these files as history events.
+Therefore when a pipeline job ceases to build successfully, you can check if something has changed on any used pipeline file.
+You can also see the build where changes occurred.
+
+image::/images/post-images/2019-07-pipeline-config-history/image2019-5-15_13-44-54.png[Screenshot of Pipeline Configuration History]
+
+Because a pipeline configuration can consist of several files where changes could have occured, only those files in which something has changed between two builds get shown at the diff.
+That makes the whole thing more compact and effective:
+
+image::/images/post-images/2019-07-pipeline-config-history/image2019-5-15_14-5-13.png[Screenshot of Pipeline Configuration History]
+
+But sometimes you may want to show not just the differences between pipeline files, also which pipeline files are in use and what the content of these files are.
+So it's possible to view all files and its contents.
+If required you can download them as well:
+
+image::/images/post-images/2019-07-pipeline-config-history/image2019-5-15_14-11-7.png[Screenshot of Pipeline Configuration History]
+
+
+== Conclusion
+
+We use Pipeline Configuration History successfully in production and it helped us from the first day on solving problems that occurred because of changes of pipeline configurations.
+Pipeline Configuration History won't replace Job Configuration History.
+Both plugins have different use cases.
+Many times small changes on job or pipeline configurations also have big impacts.
+Because of the correlation in time between changes of job or pipeline configurations and different build behavior, it is now possible to reduce the time effort to analyze the build failures substantially.
```

Comment:
```suggestion
Because of the correlation in time between changes of job or pipeline configurations and different build behavior, it is now possible to substantially reduce the time and effort to analyze build failures.
```